Automattic/jetpack
# AI Client: expose build folder

Currently, the package doesn't expose the `build/` folder in the npm repository. This PR adds the changes required to do that.
Also, it redefines the `exports` and defines the `main` and `types` package props.
Finally, it adds the composer scripts to build the code.

Fixes #

## Proposed changes:

* AI Client: expose build folder


### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:







* First of all, confirm importing the package from other projects keeps working as expected. For instance, build the JS files of the Jetpack plugin.

* Build the package locally. It will create the `/tmp/jetpack-build js-packages/ai-client` folder:

```cli
jetpack build --production --for-mirrors=/tmp/jetpack-build js-packages/ai-client
```

* go into the folder jetpack-ai-client folder:

```
cd /tmp/jetpack-build/Automattic/jetpack-ai-client
```

Create a pack defining a location. I set `~`:

```
npm pack --pack-destination ~
```

Update the dependency in your package.json by using the file location. For instance, [here](https://github.com/woocommerce/woocommerce/blob/0e174741f3c76ff4c92d1375fef20506dd4f92af/packages/js/ai/package.json#L29):

```json
"dependencies": {
	"@automattic/jetpack-ai-client": "file:~/automattic-jetpack-ai-client-0.1.16-alpha.tgz",
	//...
```

* Compile the AI Package:

```cli
pnpm run build
```
